### DISCUSSION TITLE
Ideas on Clean Architecture with Stacked (Uncle Bob) ?

### DISCUSSION THREAD

**Discussion by davidjadczyk** [FilledStacks/stacked#662] [General]
There are some articles (e.g. https://devmuaz.medium.com/flutter-clean-architecture-series-part-1-d2d4c2e75c47) discussing Uncle Bob's Clean Architecture in Flutter. This is a pretty clean way to decouple parts of the application, make it more testable, as well as restrict access to only the dependencies needed. 
I think many consumers of stacked aim to build some sort of production ready application, which most times  should keep clean scalability in mind. I personally am not experience with Clean Architecture, that's why I struggle how to structure this pattern for MVVM of Stacked in reference to StackedServices and ReactiveViewModels opposed to Blocs that are most commonly used in those Articles with Use Case classes.
I would love to see any type of ideas or example projects, that someone with more experience regarding Clean Architecture has 👍🏽 

Here a sample folder structure for CA:

![1*xR0QbSgfoODe3rH0nPwpPg](https://user-images.githubusercontent.com/35080988/172997199-828c76e7-3368-41b8-bdd6-4380eb94e526.png)

Here a sample flow for CA:
![1*ggdsjSaIY2Q_PHG-Db0BVQ](https://user-images.githubusercontent.com/35080988/172997530-7cfa4530-6638-4940-96b2-e843f267b8ad.png)

**Comment by FilledStacks** [comment]
I don't think adding more layers on Stacked would benefit the codebase much. It's quite a bit of boilerplate and my goal is to create the simplest form of a scalable and maintainable code base. CLEAN architecture is on the other side of our direction at the moment. but I'm sure it can be done.
